### ISSUE TITLE
Failed to transform artifact "fingerprintidentify.aar" (androidx support)

### ISSUE THREAD

**Issue by andrey-shostik** [hieuvp/react-native-fingerprint-scanner#88]
I can't build android app. I receive the error https://paste.ofcode.org/4d53TWRmreQ7NTLr46iVX5
I started to receive this error after adding following to `gradle.properties`
```
android.useAndroidX=true
android.enableJetifier=true
```
my react-native config
```
React Native Environment Info:
    System:
      OS: macOS 10.14.5
      CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
      Memory: 2.39 GB / 16.00 GB
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 10.16.0 - /usr/local/bin/node
      Yarn: 1.12.3 - /usr/local/bin/yarn
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
      Android SDK:
        API Levels: 23, 25, 26, 27, 28
        Build Tools: 23.0.1, 25.0.3, 26.0.2, 26.0.3, 27.0.3, 28.0.2, 28.0.3
        System Images: android-24 | Google Play Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom_64
    IDEs:
      Android Studio: 3.1 AI-173.4907809
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.8.3 => 16.8.3 
      react-native: 0.59.9 => 0.59.9 
    npmGlobalPackages:
      react-native-cli: 2.0.1
```

**Comment by amandamussio** [comment]
Same problem here!

**Comment by lucky10** [comment]
Same issue

**Comment by phillbaker** [comment]
I suspect this is because the underlying android library doesn't support androidx until a version after what's specified on the currently released version.

Can you try adding to your `android/gradle.properties`:
```
fingerprintidentify=1.2.6
```

See https://github.com/hieuvp/react-native-fingerprint-scanner/issues/86#issuecomment-497156584 for the inverse issue.

**Comment by phillbaker** [comment]
@lucky10 @amandamussio @andrey-shostik did the above fix work for anyone?

**Comment by amandamussio** [comment]
> @lucky10 @amandamussio @andrey-shostik did the above fix work for anyone?

Not's worked for me. :(

**Comment by praveens96** [comment]
getting same issue:
Could not resolve all files for configuration ':app:debugCompileClasspath'.
> Failed to transform file 'fingerprintidentify-1.2.1.aar' to match attributes {artifactType=processed-aar} using transform JetifyTransform
   > Failed to transform 'xxx\.gradle\caches\modules-2\files-2.1\com.wei.android.lib\fingerprintidentify\1.2.1\4d29dc54b0145ba76535b98cea61a4dc4a714fc4\fingerprintidentify-1.2.1.aar' using Jetifier. Reason: null. (Run with --stacktrace for more details.) To disable Jetifier, set android.enableJetifier=false in your gradle.properties file.

**Comment by ilotte99** [comment]
same issue with me

**Comment by phillbaker** [comment]
Hey all - just to clarify, my understanding is that react-native v0.60 is the first one compatible with androidx? The release notes for 0.60 seem to indicate that too: https://facebook.github.io/react-native/blog/2019/07/03/version-60#androidx-support

**Comment by phillbaker** [comment]
Hi all, I've pushed initial support for react-native v0.60 and androidx on a branch, it's here: https://github.com/hieuvp/react-native-fingerprint-scanner/tree/update/rn-60

If you'd like to try it out, you can update your package.json to point to it via:
```
"react-native-fingerprint-scanner": "github:hieuvp/react-native-fingerprint-scanner#7515eac",
```

With the changes to native dependencies, I had to "unlink" the android side, but not the iOS. The diff includes changes to the example project which compiles and runs. Specifically, I had to comment out `packages.add(new ReactNativeFingerprintScannerPackage());` in `MainApplication`'s `getPackages()`. I also included a 

Remaining items:
- [ ] bump major version
- [ ] include notice in readme with instructions, a la https://github.com/transistorsoft/react-native-background-fetch#ios-setup

**Comment by Thomazella** [comment]
Guy from #89 here. My react-native-info:
```
rn info
info Fetching system and libraries information...
System:
    OS: macOS 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
    Memory: 3.97 GB / 16.00 GB
    Shell: 5.7.1 - /usr/local/bin/zsh
  Binaries:
    Node: 12.5.0 - ~/.nvm/versions/node/v12.5.0/bin/node
    Yarn: 1.17.0 - /usr/local/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v12.5.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
    Android SDK:
      API Levels: 23, 25, 26, 27, 28
      Build Tools: 23.0.1, 23.0.2, 26.0.1, 26.0.2, 26.0.3, 27.0.2, 27.0.3, 28.0.0, 28.0.3, 29.0.0
      System Images: android-19 | Google APIs Intel x86 Atom, android-22 | Google APIs Intel x86 Atom, android-23 | Google APIs Intel x86 Atom, android-26 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5522156
    Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6
    react-native: 0.60.0 => 0.60.0
```

**Comment by Thomazella** [comment]
This has unblocked me and I've successfully compiled & ran my app on react-native 0.60 💯

**Comment by phillbaker** [comment]
Thanks all - going to re-open so that we can track progress until the above work is merged.

**Comment by Thomazella** [comment]
@phillbaker by `figure out link/unlink upgrade path for ios vs. android` do you mean write a guide?

**Comment by phillbaker** [comment]
CLosed with 7e828060e30274c6def68bf5b01dd8af28954586
